lib/src/main/kotlin/com/swmansion/starknet/deployercontract/Deployer.kt

```diff
@@ -35,8 +36,43 @@ interface Deployer {
         unique: Boolean,
         salt: Felt,
         constructorCalldata: Calldata,
+        maxFee: Felt,
     ): Request<ContractDeployment>
 
+    /**
+     * Deploy a contract through Universal Deployer Contract (UDC)
+     *
+     * @param classHash a class hash of the declared contract
+     * @param unique set whether deployed contract address should be based on account address or not
+     * @param salt a salt to be used to calculate deployed contract address
+     * @param constructorCalldata constructor calldata
+     *
+     * @throws RequestFailedException
+     */
+    fun deployContract(
+        classHash: Felt,
+        unique: Boolean,
+        salt: Felt,
+        constructorCalldata: Calldata,
+    ): Request<ContractDeployment>
+
+    /**
+     * Deploy a contract through Universal Deployer Contract (UDC) with random generated salt and
+     * unique parameter set to true
+     *
+     * @param classHash a class hash of the declared contract
+     * @param constructorCalldata constructor calldata
+     * @param maxFee maximum fee that account will use for the deployment
+     * @throws RequestFailedException
+     * @throws SaltGenerationFailedException
+     */
+    fun deployContract(classHash: Felt, constructorCalldata: Calldata, maxFee: Felt): Request<ContractDeployment> {
+        val random = SecureRandom()
+        val salt = random.longs(1, 1, Long.MAX_VALUE).findFirst().orElseThrow { SaltGenerationFailedException() }
+        val feltSalt = Felt(salt)
+        return deployContract(classHash, true, feltSalt, constructorCalldata, maxFee)
+    }
+
```

Comment:
let's keep implementations away from this file. btw, do we really need this one?

Reply:
Just below, the same approach was used. 
But I agree we should keep implementations away from interfaces for consistency.

As for the reasons why we need deployContract with maxFee parameter, I tried to outline those in #361

Reply:
I meant this specific implementation - you have already added max_fee to deployContract at the top - why create new variation (with implied uniqueness and random salt)?

Reply:
There were previously two implementations, one with all the all the parameters, and the other one without the parameters that can be assumed, so I thought it would be useful to allow passing maxFee in both cases.

I can imagine someone wanting to not bother with generating random salt while also passing maxFee at the same time.

Reply:
meh, maybe. that's a detail though, if you want to leave this that's fine